snuba/request/schema.py

```diff
@@ -62,15 +64,17 @@ def __init__(
 
     @classmethod
     def build_with_extensions(
-        cls, extensions: Mapping[str, QueryExtension]
+        cls,
+        extensions: Mapping[str, QueryExtension],
+        settings_class: Type[RequestSettings] = HTTPRequestSettings,
     ) -> RequestSchema:
         generic_schema = GENERIC_QUERY_SCHEMA
         settings_schema = SETTINGS_SCHEMA
```

Comment:
Thanks, this is a good direction to take this. I could see a case where having `debug` available might be useful during dev, just in terms of seeing exactly what query was run to produce the result. I don't think it's worth including now though, I'll add it in if I end up needing it. Then I think that the `debug` setting would be stored on my end and passed via the api (not sure if this makes sense since we're typically not mutating subscriptions).

Reply:
Makes sense — let's cross that bridge when we come to it. My suspicion is we'll probably need something out of band to control turning on debug for some set of queries (all of them? within a project? by an ID? I'm not sure, specifically) since like you mentioned we aren't/shouldn't be mutating them.